pubnub-kotlin-core/core-impl/src/main/kotlin/com/pubnub/internal/PNConfiguration.kt

```diff
@@ -0,0 +1,423 @@
+         *
+         * @see [Proxy]
+         */
+        var proxy: Proxy? = null
+
+        /**
+         * @see [ProxySelector]
+         */
+        var proxySelector: ProxySelector? = null
+
+        /**
+         * @see [Authenticator]
+         */
+        var proxyAuthenticator: Authenticator? = null
+
+        /**
+         * @see [CertificatePinner]
+         */
+        var certificatePinner: CertificatePinner? = null
+
+        /**
+         * Sets a custom [HttpLoggingInterceptor] for logging network traffic.
+         *
+         * @see [HttpLoggingInterceptor]
+         */
+        var httpLoggingInterceptor: HttpLoggingInterceptor? = null
+
+        /**
+         * @see [SSLSocketFactory]
+         */
+        var sslSocketFactory: SSLSocketFactory? = null
+
+        /**
+         * @see [X509ExtendedTrustManager]
+         */
+        var x509ExtendedTrustManager: X509ExtendedTrustManager? = null
+
+        /**
+         * @see [okhttp3.ConnectionSpec]
+         */
+        var connectionSpec: ConnectionSpec? = null
+
+        /**
+         * @see [javax.net.ssl.HostnameVerifier]
+         */
+        var hostnameVerifier: HostnameVerifier? = null
+
+        /**
+         * How many times publishing file message should automatically retry before marking the action as failed
+         *
+         * Defaults to `5`
+         */
+        var fileMessagePublishRetryLimit = 5
+
+        var dedupOnSubscribe = false
+        var maximumMessagesCacheSize = DEFAULT_DEDUPE_SIZE
+
+        val pnsdkSuffixes: ConcurrentMap<String, String> = ConcurrentHashMap(mutableMapOf())
+
+        fun generatePnsdk(version: String): String {
+            val joinedSuffixes = pnsdkSuffixes.toSortedMap().values.joinToString(" ")
+            return "PubNub-Kotlin/$version" + if (joinedSuffixes.isNotBlank()) " $joinedSuffixes" else ""
+        }
+
+        @Suppress("DeprecatedCallableAddReplaceWith")
+        @Deprecated("To be used by components", level = DeprecationLevel.WARNING)
+        fun addPnsdkSuffix(vararg nameToSuffixes: Pair<String, String>) {
+            @Suppress("DEPRECATION")
+            addPnsdkSuffix(nameToSuffixes.toMap())
+        }
+
+        @Deprecated("To be used by components", level = DeprecationLevel.WARNING)
+        fun addPnsdkSuffix(nameToSuffixes: Map<String, String>) = pnsdkSuffixes.putAll(nameToSuffixes)
+
+        /**
+         * Retry configuration for requests.
+         *  Defaults to [RetryConfiguration.None].
+         *
+         *  Use [RetryConfiguration.Linear] to set retry with linear delay interval
+         *  Use [RetryConfiguration.Exponential] to set retry with exponential delay interval
+         *  Delay will vary from provided value by random value.
+         */
+        var retryConfiguration: RetryConfiguration = RetryConfiguration.None
+
+        internal val retryConfForOldSubscribeLoop: RetryConfiguration by lazy {
```

Comment:
I guess this property can be deleted since we don't have old subscribeLoop code any more.

Reply:
done